File: CreationalPatterns/Prototype/Java/Implementations/MargheritaPizza.java
package Implementations;

import Interfaces.AbstractPizza;
import Prototype.IPizzaPrototype;

public class MargheritaPizza extends AbstractPizza {

    public MargheritaPizza(){
        super();
        this.dough = "Hand Tossed";
        this.sauce = "Tomato";
        this.toppings.add("Mozzarella Cheese");
    }

    public MargheritaPizza(MargheritaPizza source){
        super(source);
    }

    @Override
    public IPizzaPrototype copy() {
        return new MargheritaPizza(this);
    }    
}

File: CreationalPatterns/Prototype/C#/Prototype/Implementations/MargheritaPizza.cs
public class MargheritaPizza : AbstractPizza {

    public MargheritaPizza(){
        this.Dough = "Hand Tossed";
        this.Sauce = "Tomato";
        this.Toppings.Add("Mozzarella Cheese");
    }

    public MargheritaPizza(MargheritaPizza source) : base(source){
    }

    public override IPizzaPrototype Copy() {
        return new MargheritaPizza(this);
    }    
}
